module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,14 @@
+#include "sort.hpp"
+
+#include <algorithm>
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> data) {
+    data.sort([](const std::string& lhs, const std::string& rhs) {
+        return (lhs.size() == rhs.size()) ? (lhs < rhs) : (lhs.size() < rhs.size());
+    });
+
+    std::deque<std::string> result;
+    std::copy(data.begin(), data.end(), std::back_inserter(result));
+
+    return result;
```

Comment:
You can  simplify it by use **std::deque<T>( data.begin(), data.end() )** constructor at return line.